### ISSUE TITLE
0.10.0 doesn't compile, latest version does. Downgrade for clboss possible or required?

### ISSUE THREAD

**Issue by GhostTyper** [ElementsProject/lightning#4702]
### Issue and Steps to Reproduce
When I download and build Release 0.10.0 on my debian 10 box I get this:

`./configure` and `make`:

```
cc ccan/ccan/time/time.c
cc ccan/ccan/timer/timer.c
cc ccan/ccan/utf8/utf8.c
cd external/libwally-core && ./tools/autogen.sh
/bin/sh: 1: ./tools/autogen.sh: not found
make: *** [external/Makefile:72: external/x86_64-linux-gnu/libwally-core-build/src/libwallycore.la] Error 127
```

Repeating the compile operation or using `make -j 1` doesn't change anything. However, compiling the latest (checked out some days ago, `v0.10.1rc2-1-ga8b3a1c`) is working just fine.

### `getinfo` output
```
{
   "id": "039306fa4008865638e08d3f43eeee43c0a6d23ed1571e88a17ef384cbd5be656e",
   "alias": "GhostTyper",
   "color": "4080ff",
   "num_peers": 6,
   "num_pending_channels": 0,
   "num_active_channels": 4,
   "num_inactive_channels": 0,
   "address": [
      {
         "type": "ipv4",
         "address": "81.95.4.21",
         "port": 9735
      }
   ],
   "binding": [
      {
         "type": "ipv4",
         "address": "81.95.4.21",
         "port": 9735
      }
   ],
   "version": "v0.10.1rc2-1-ga8b3a1c",
   "blockheight": 694883,
   "network": "bitcoin",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/home/btc/.lightning/bitcoin"
}
```

### Why I opened this issue at all.
I now opened channels, etc. and thought that I may want to try clboss. But I'm afraid now, that it may be incompatible to this interim version. My next thoughts have been, that it also may be a problem upgrading this interim version to the next future release version. And then additionally my thoughts were that it also may be a problem downgrading to 0.10.0, if I get it compiled.

My Question is: Do I need to close all channels and try again with a fresh start? Or is it safe to say that all versions of 0.10.X use the same database format, etc. and it is therefore safe to downgrade or later upgrade? Maybe clboss will just work with this interim version?

If I have to use a release version (0.10.0 in that case): What Do I need to make this version compile on debian 10 when the current version in the repository compiles on my system?

**Comment by whitslack** [comment]
CLBoss will work with the 0.10.1 release candidate. If there were any backward-incompatible changes, the version number would be bumped to 0.11.

It is not safe to say that all 0.10.X versions use the same database format. There are several database migrations in 0.10.1. You should not run 0.10.0 on the same database after running any 0.10.1 release candidate, or you risk getting your database into an inconsistent state that can't automatically be fixed when you upgrade to 0.10.1.

**Comment by GhostTyper** [comment]
Thank you very much, @whitslack.
